contracts/src/v0.8/dev/KeeperRegistryDev.sol

```diff
@@ -401,92 +391,67 @@ contract KeeperRegistryDev is
    * @dev the order of IDs in the list is **not guaranteed**, therefore, if making successive calls, one
    * should consider keeping the blockheight constant to ensure a wholistic picture of the contract state
    */
-  function getActiveUpkeepIDs(uint256 startIndex, uint256 maxCount) external view override returns (uint256[] memory) {
-    uint256 maxIdx = s_upkeepIDs.length();
-    if (startIndex >= maxIdx) revert IndexOutOfRange();
-    if (maxCount == 0) {
-      maxCount = maxIdx - startIndex;
-    }
-    uint256[] memory ids = new uint256[](maxCount);
-    for (uint256 idx = 0; idx < maxCount; idx++) {
-      ids[idx] = s_upkeepIDs.at(startIndex + idx);
-    }
-    return ids;
+  function getActiveUpkeepIDs(uint256 startIndex, uint256 maxCount) external override returns (uint256[] memory) {
+    // Executed through logic contract
+    _fallback();
   }
 
   /**
    * @notice read the current info about any keeper address
    */
   function getKeeperInfo(address query)
     external
-    view
     override
     returns (
       address payee,
       bool active,
       uint96 balance
     )
   {
-    KeeperInfo memory keeper = s_keeperInfo[query];
-    return (keeper.payee, keeper.active, keeper.balance);
+    // Executed through logic contract
+    _fallback();
   }
 
   /**
    * @notice read the current state of the registry
    */
   function getState()
     external
-    view
     override
     returns (
       State memory state,
       Config memory config,
       address[] memory keepers
     )
   {
-    Storage memory store = s_storage;
-    state.nonce = store.nonce;
-    state.ownerLinkBalance = s_ownerLinkBalance;
-    state.expectedLinkBalance = s_expectedLinkBalance;
-    state.numUpkeeps = s_upkeepIDs.length();
-    config.paymentPremiumPPB = store.paymentPremiumPPB;
-    config.flatFeeMicroLink = store.flatFeeMicroLink;
-    config.blockCountPerTurn = store.blockCountPerTurn;
-    config.checkGasLimit = store.checkGasLimit;
-    config.stalenessSeconds = store.stalenessSeconds;
-    config.gasCeilingMultiplier = store.gasCeilingMultiplier;
-    config.minUpkeepSpend = store.minUpkeepSpend;
-    config.maxPerformGas = store.maxPerformGas;
-    config.fallbackGasPrice = s_fallbackGasPrice;
-    config.fallbackLinkPrice = s_fallbackLinkPrice;
-    config.transcoder = s_transcoder;
-    config.registrar = s_registrar;
-    return (state, config, s_keeperList);
+    // Executed through logic contract
+    _fallback();
   }
 
   /**
    * @notice calculates the minimum balance required for an upkeep to remain eligible
    * @param id the upkeep id to calculate minimum balance for
    */
-  function getMinBalanceForUpkeep(uint256 id) external view returns (uint96 minBalance) {
-    return getMaxPaymentForGas(s_upkeep[id].executeGas);
+  function getMinBalanceForUpkeep(uint256 id) external returns (uint96 minBalance) {
+    // Executed through logic contract
+    _fallback();
```

Comment:
Oh boy... I forgot you can't use delegate call inside a view function. I'm worried this is going to make debugging a nightmare and a half.

Reply:
added `cannotExecute` to these functions and I think that's the best we can do if we want to move these functions over to logic. 

I'll put up another PR to move admin functions to logic and see the size savings there. We can then take a decision on the order, though I suspect we'll need both